lib/process/process.go

```diff
@@ -1706,7 +1706,10 @@ func (p *Process) initClusterCertificate(client *kubernetes.Clientset) error {
 		return trace.Wrap(err)
 	}
 
-	_, err = p.operator.UpdateClusterCertificate(ops.UpdateCertificateRequest{
+	localCtx := context.WithValue(ctx, constants.UserContext,
```

Comment:
Why is this not a direct dependency for `UpdateClusterCertificate` but instead is being passed on via context?

Reply:
UpdateClusterCertificate can be called by a user (e.g. when a user updates cert via UI) and in that case the context will contain the appropriate authenticated user. In this case, there is no external authenticated "user" per se - the process itself is a user, so it passes itself (`@system`) in the context.